### PR TITLE
[frontend|backend]Show prerequisites info in atomic testing (#1787)

### PR DESCRIPTION
**Pull request by johanah29**
#1856

### CODE REVIEW

File: openbas-api/src/main/java/io/openbas/rest/payload/PayloadUtils.java [hunk 1]
```diff
@@ -40,8 +40,8 @@ public static PayloadCreateInput buildPayload(@NotNull final JsonNode payloadNod
       payloadCreateInput.setContent(payloadNode.get("command_content").textValue());
     }
     if (payloadNode.has("executable_arch")) {
-      payloadCreateInput.setExecutableArch(
-          Endpoint.PLATFORM_ARCH.valueOf(payloadNode.get("executable_arch").textValue()));
+      payloadCreateInput.setExecutionArch(
+          Payload.PAYLOAD_EXECUTION_ARCH.valueOf(payloadNode.get("executable_arch").textValue()));
     }
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/rest/payload/PayloadUtils.java:45]: if (payloadNode.has("payload_execution_arch")) {
..
payloadNode.get("payload_execution_arch")

File: openbas-api/src/main/java/io/openbas/scheduler/jobs/InjectsExecutionJob.java [hunk 2]
```diff
@@ -200,7 +200,7 @@ private void executeInject(ExecutableInject executableInject) {
               errorMsg -> finalStatus.getTraces().add(InjectStatusExecution.traceError(errorMsg)));
       finalStatus.setName(ExecutionStatus.ERROR);
       finalStatus.setTrackingSentDate(Instant.now());
-      finalStatus.setCommandsLines(injectUtils.getCommandsLinesFromInject(inject));
+      finalStatus.setPayloadOutput(injectUtils.getCommandsLinesFromInject(inject));
       injectStatusRepository.save(finalStatus);
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/scheduler/jobs/InjectsExecutionJob.java:204]: InjectUtils.getCommandsLinesFromInject maybe should be : InjectUtils.getPaylodOutput or getStatusPayloadOutput

File: openbas-api/src/main/java/io/openbas/utils/InjectUtils.java [hunk 3]
```diff
@@ -29,37 +33,124 @@ public class InjectUtils {
 
   private final ApplicationContext context;
 
-  public InjectStatusCommandLine getCommandsLinesFromInject(final Inject inject) {
+  public StatusPayload getCommandsLinesFromInject(final Inject inject) {
     if (inject == null) {
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/InjectUtils.java:37]: getStatusPayload

File: openbas-api/src/main/java/io/openbas/utils/InjectUtils.java [hunk 4]
```diff
@@ -29,37 +33,124 @@ public class InjectUtils {
...
-            payloadCommand.getCleanupCommand() != null
-                    && !payloadCommand.getCleanupCommand().isBlank()
-                ? List.of(payload.getCleanupCommand())
-                : null,
-            payload.getExternalId());
+        PayloadCommandBlock payloadCommandBlock =
+            new PayloadCommandBlock(
+                payloadCommand.getExecutor(),
+                payloadCommand.getContent(),
+                List.of(payloadCommand.getCleanupCommand()));
```
Comment (savacano28) [inline openbas-api/src/main/java/io/openbas/utils/InjectUtils.java:55]: getCleanCommand can be null? NPE

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingDetail.tsx [hunk 5]
```diff
@@ -33,130 +24,14 @@ const useStyles = makeStyles(() => ({
...
 const AtomicTestingDetail: FunctionComponent<Props> = () => {
   const classes = useStyles();
-  const { t, tPick } = useFormatter();
-  const dispatch = useAppDispatch();
+  const { t } = useFormatter();
 
   // Fetching data
   const { injectResultOverviewOutput } = useContext<InjectResultOverviewOutputContextType>(InjectResultOverviewOutputContext);
-  const { documentMap } = useHelper((helper: DocumentHelper) => ({
-    documentMap: helper.getDocumentsMap(),
```
Comment (savacano28) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/AtomicTestingDetail.tsx:42]: Are the docs removed from Atomic UI?

File: openbas-api/src/main/java/io/openbas/utils/PayloadMapper.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+              .cleanupExecutor(payload.getCleanupExecutor())
+              .name(injectorContract.getPayload().getName())
+              .type(injectorContract.getPayload().getType())
+              .collectorType(injectorContract.getPayload().getCollectorType())
+              .description(injectorContract.getPayload().getDescription())
+              .platforms(injectorContract.getPayload().getPlatforms())
+              .attackPatterns(toAttackPatternSimples(injectorContract.getAttackPatterns()))
+              .executableArch(injectorContract.getArch());
+          if (COMMAND_TYPE.equals(injectorContract.getPayload().getType())) {
+            Command payloadCommand = (Command) Hibernate.unproxy(payload);
```
[the remark this answers is not in the document]
Comment (savacano28) [reply openbas-api/src/main/java/io/openbas/utils/PayloadMapper.java:45]: there are more lines to replace

File: openbas-front/src/admin/components/atomic_testings/atomic_testing/Index.tsx [hunk 7]
```diff
@@ -106,12 +107,24 @@ const Index = () => {
                 label={t('Execution details')}
                 className={classes.item}
               />
+              {
+                injectResultOverviewOutput.inject_type !== 'openbas_email' && injectResultOverviewOutput.inject_type !== 'openbas_ovh_sms' && (
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/atomic_testings/atomic_testing/Index.tsx:111]: You missed the Mastodon Inject
image

**Review by savacano28 — APPROVED** [review]

**Pull request merged**